persist/fs/commitlog/types.go

```diff
@@ -105,8 +105,24 @@ type Series struct {
 
 	// Shard is the shard the series belongs to
 	Shard uint32
+
+	// SegmentFlags declares whether to finalize when finalizing the Series.
+	Flags SeriesFlags
 }
 
+// SeriesFlags describes the option to finalize or not finalize
+// bytes in a Series.
+type SeriesFlags uint8
+
+const (
```

Comment:
This way of assigning flag values is wrong, it should be 1 << 0, 1 << 1, 1 << 2 and so on. Otherwise you'll end up with overlapping values, e.g. 3 << 1 == 6, which is 0b00000110 – note two 1s.